sources/xfscli/src/bin/fsck_xfs.rs

```diff
@@ -0,0 +1,169 @@
+                    exit(1);
+                }
+
+                let params =
+                    fs::read_to_string("/proc/cmdline").expect("Unable to read /proc/cmdline file");
+                let mut root = String::new();
+                let mut root_flags = "-o ".to_string();
+                for param in params.split(' ') {
+                    if param.starts_with("root=") {
+                        root = param.strip_prefix("root=").unwrap().to_string();
+                    }
+                    if param.starts_with("rootflags=") {
+                        root_flags = root_flags
+                            .clone()
+                            .add(param.strip_prefix("rootflags=").unwrap());
+                    }
+                }
+                let path = Path::new(&root);
+                let metadata = path.metadata().expect("Could not fetch metadata");
+                let file_type = metadata.file_type();
+
+                if !file_type.is_block_device() {
+                    let blkid_result = Command::new(BLKID)
+                        .args(["-t", &root, "-o", target])
+                        .output()
+                        .expect("Couldn't run blkid");
+                    root = String::from_utf8(blkid_result.stdout).unwrap();
+                }
+
+                let basename_dev = Path::new(target).file_name().unwrap();
+
+                let basename_root = Path::new(&root).file_name().unwrap();
+
+                let mut mount_args = vec![target.as_str(), "/tmp/repair_mnt"];
+                if basename_dev == basename_root {
+                    mount_args.push(root_flags.as_str());
+                }
+
+                let mount_result = Command::new(MOUNT)
+                    .args(mount_args)
+                    .output()
+                    .expect("Couldn't run mount");
+                if !mount_result.status.success() {
+                    exit(1);
+                }
+
+                Command::new(UMOUNT)
+                    .args(["/tmp/repair_mnt"])
+                    .output()
+                    .expect("Couldn't run unmount");
+
+                println!("Performing XFS repair again after cleaning the log..");
+                repair_exit_code = perform_xfs_repair(vec!["-e", target]);
+                fs::remove_dir("/tmp/repair_mnt")
+                    .expect("could not delete the /tmp/repair_mnt dir");
+            }
+            exit(repair2fsck_code(repair_exit_code));
+        }
+
+        if auto {
+            println!("fsck.xfs : XFS file system.");
+        } else {
+            println!(
+                "If you wish to check the consistency of an XFS filesystem or \
+            repair a damaged filesystem, see xfs_repair(8)."
+            );
+        }
+        exit(0);
+    } else {
+        println!("Couldn't parse target block-device ");
+    }
+}
+
+fn repair2fsck_code(error: i32) -> i32 {
+    match error {
+        0 => 0, // everything is ok
```

Comment:
It would be better to give meaningful names to each exit code, probably with an ENUM. And the same for the returned values.